### ISSUE TITLE
[glib-static(gtkmm)] build failure

### ISSUE THREAD

**Issue by OptimizedIAS** [microsoft/vcpkg#10589]
**Host Environment**
- OS: Windows 10 64 bit
- Compiler: Vs2019 compiler (Microsoft (R) C/C++ Optimizing Compiler Version 19.24.28315)

**To Reproduce**
`./vcpkg install gtkmm:x86-windows-static

**Failure logs** 
Starting package 1/21: glib:x86-windows-static
Building package glib[core]:x86-windows-static...
-- Using community triplet x86-windows-static. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: C:\Users\Optimizer\Downloads\Compressed\vcpkg-master\vcpkg-master\triplets\community\x86-windows-static.cmake
-- Note: glib only supports dynamic library linkage. Building dynamic library.
CMake Error at scripts/cmake/vcpkg_check_linkage.cmake:42 (message):
  Refusing to build unexpected dynamic library against the static CRT.  If
  this is desired, please configure your triplet to directly request this
  configuration.
Call Stack (most recent call first):
  ports/glib/portfile.cmake:6 (vcpkg_check_linkage)
  scripts/ports.cmake:90 (include)

Error: Building package glib:x86-windows-static failed with: BUILD_FAILED

**Additional context**

I understand that glib can't be build as a static lib on windows. I wanna know how to do this.

"Refusing to build unexpected dynamic library against the static CRT.  If
  this is desired, **please configure your triplet to directly request this
  configuration**"

Basically, I want to build gtkmm as a static library while allowing dependencies that should be built as dynamic, to do so.

**Comment by Voskrese** [comment]
-- Note: glib only supports dynamic library linkage. Building dynamic library.

**Comment by LilyWangL** [comment]
Thanks for posting issue. Port gtkmm doesn't support static build for now, it will will come true in the future.
